test/unit/org/apache/cassandra/db/virtual/GossipStateTransitionsTableTest.java

```diff
@@ -0,0 +1,96 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.cassandra.db.virtual;
+
+import com.google.common.collect.ImmutableList;
+import org.junit.Before;
+import org.junit.BeforeClass;
+import org.junit.Test;
+
+import org.apache.cassandra.config.DatabaseDescriptor;
+import org.apache.cassandra.cql3.CQLTester;
+import org.apache.cassandra.cql3.UntypedResultSet;
+import org.apache.cassandra.gms.Gossiper;
+import org.apache.cassandra.utils.FBUtilities;
+
+import static org.apache.cassandra.utils.Clock.Global.currentTimeMillis;
+import static org.assertj.core.api.Assertions.assertThat;
+
+/**
+ * Tests for the {@link GossipStateTransitionsTable} virtual table.
+ */
+public class GossipStateTransitionsTableTest extends CQLTester
+{
+    private static final String KS_NAME = "vts";
+
+    @SuppressWarnings("FieldCanBeLocal")
+    private VirtualTable table;
+
+    @BeforeClass
+    public static void setup() throws Exception
+    {
+        // limit the transitions to record to 2 per host
+        DatabaseDescriptor.setMaxGossipStateTransitionsSize(2);
+        setUpClass();
+        startJMXServer();
+    }
+
+    @Before
+    public void config()
+    {
+        table = new GossipStateTransitionsTable(KS_NAME);
```

Comment:
unsafe pattern; is there a reason you need to keep creating the table? I had to "fix"/"work around" this pattern in https://issues.apache.org/jira/browse/CASSANDRA-17295

A safer pattern is to set this up in the `setup` method, you also don't need to save the table to this class